### PR TITLE
stmts: Fix bug in update single-table syntax but with multiple tables

### PR DESCRIPTION
**Pull request by shenli**
Fix bug found in beego/orm

### CODE REVIEW

File: rset/rsets/join.go [hunk 1]
```diff
@@ -231,3 +231,13 @@ func (r *JoinRset) Plan(ctx context.Context) (plan.Plan, error) {
 
 	return p, nil
 }
+
+// MultipleTable checks if the JoinRset contains multiple tables
```
Comment (qiuyesuifeng) [inline rset/rsets/join.go]: End of .

File: rset/rsets/join.go [hunk 2]
```diff
@@ -231,3 +231,13 @@ func (r *JoinRset) Plan(ctx context.Context) (plan.Plan, error) {
...
+func (r *JoinRset) MultipleTable() bool {
+	if _, ok := r.Left.(*TableSource); ok {
```
Comment (qiuyesuifeng) [inline rset/rsets/join.go]: If `r.Left` is not `*TableSource`, can this work well?

File: stmt/stmts/update.go [hunk 3]
```diff
@@ -241,6 +241,8 @@ func (s *UpdateStmt) Exec(ctx context.Context) (_ rset.Recordset, err error) {
 	}
 	defer p.Close()
 	updatedRowKeys := make(map[string]bool)
+	// For single-table syntax, TableRef may contain multiple tables
+	multipleTables := s.MultipleTable || s.TableRefs.MultipleTable()
```
Comment (qiuyesuifeng) [inline stmt/stmts/update.go]: How about `isMultipleTables` variable name?

File: rset/rsets/join.go (hunk 1 above)
Comment (shenli) [inline rset/rsets/join.go]: PTAL

File: stmt/stmts/update.go [hunk 4]
```diff
@@ -85,19 +85,19 @@ func (s *UpdateStmt) SetText(text string) {
...
 	tname := t.TableName()
 	for _, asgn := range assignList {
-		if multipleTable {
+		if isMultipleTable {
 			if !strings.EqualFold(tname.O, asgn.TableName) {
 				continue
 			}
 		}
 		col := column.FindCol(t.Cols(), asgn.ColName)
 		if col == nil {
```
Comment (siddontang) [inline stmt/stmts/update.go]: why isMutipleTable skips this error?

Comment (shenli) [inline stmt/stmts/update.go]: The TableRefs may contains multiple table. So not all the column mentioned in assignList are in table t.

**Comment by siddontang** [comment]
LGTM

**Pull request merged**